internal/sci/aws/server.go

```diff
@@ -0,0 +1,285 @@
+// Package aws provides an AWS implementation of the Substratus Cloud Interface (SCI)
+package aws
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	awsSdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/substratusai/substratus/internal/sci"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Server struct {
+	sci.UnimplementedControllerServer
+	OIDCProviderURL string
+	OIDCProviderARN string
+	Clients
+}
+
+type Clients struct {
+	S3Client  *s3.S3
+	IamClient *iam.IAM
+}
+
+func NewAWSServer() (*Server, error) {
```

Comment:
Convention is to construct clients in main() and pass in to make it very clear what a program is depending on. Typically best to avoid the combo of BOTH configuring AND constructing clients from within pkg functions.

Reply:
Rename `NewServer` because the struct name is `Server`

Reply:
Done